src/serve/hosting.ts

```diff
@@ -9,22 +10,59 @@ import { normalizedHostingConfigs } from "../hosting/normalizedHostingConfigs";
 import * as utils from "../utils";
 import cloudRunProxy from "../hosting/cloudRunProxy";
 import functionsProxy from "../hosting/functionsProxy";
+import { NextFunction, Request, Response } from "express";
+import { Writable } from "stream";
 
 const MAX_PORT_ATTEMPTS = 10;
 let attempts = 0;
 let server: any; // eslint-disable-line @typescript-eslint/no-explicit-any
 
+const logData = {
+  metadata: {
+    emulator: {
+      name: "hosting",
+    },
+  },
+};
+
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 function startServer(options: any, config: any, port: number, init: TemplateServerResponse): void {
+  const firebaseMiddleware = initMiddleware(init);
+
+  const morganStream = new Writable();
+  morganStream._write = (
```

Comment:
I really don't like touching private functions of external libraries.  Could we do something cleaner if we updated superstatic?

Reply:
implementing `_write` is literally how you make a custom writeable stream, it's just an ancient NodeJS api. We could also update superstatic, but we'd do the exact same thing, but move the code from here to superstatic